app/rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationRevisionHandler.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+
+    @Path("/{id}/{version}")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public IntegrationRevision get(@NotNull @PathParam("id") @ApiParam(required = true) String id, @NotNull @PathParam("version") @ApiParam(required = true) String version) {
+        String compositeId = id +"#" + version;
+        IntegrationRevision integrationRevision = Getter.super.get(compositeId);
+
+        if (integrationRevision.getCurrentState() == IntegrationRevisionState.Undeployed ||
+            integrationRevision.getTargetState() == IntegrationRevisionState.Undeployed) {
+            //Not sure if we need to do that for both current and desired status,
+            //but If we don't do include the desired state, IntegrationITCase is not going to pass anytime soon. Why?
+            //Cause that test, is using NoopHandlerProvider, so that means no controllers.
+            throw new EntityNotFoundException(String.format("Integration revision %s has been deleted", integrationRevision.getId()));
+        }
+
+        //fudging the timesUsed for now
+        if (integrationRevision.getCurrentState() == IntegrationRevisionState.Active) {
+            return new IntegrationRevision.Builder().createFrom(integrationRevision)
+                .timesUsed(BigInteger.valueOf(new Date().getTime()/1000000))
+                .build();
+        }
+
+        return integrationRevision;
+    }
+
+
+    @Override
+    public ListResult<IntegrationRevision> list(UriInfo uriInfo) {
+        return getDataManager().fetchAll(
+            IntegrationRevision.class,
+            new IntegrationRevisionHandler.DeletedFilter(),
+            new ReflectiveSorter<>(IntegrationRevision.class, new SortOptionsFromQueryParams(uriInfo)),
+            new PaginationFilter<>(new PaginationOptionsFromQueryParams(uriInfo))
+        );
+    }
+
+    @Override
+    public IntegrationRevision create(@Context SecurityContext sec, @ConvertGroup(from = Default.class, to = AllValidations.class) final IntegrationRevision integrationRevision) {
+        Date rightNow = new Date();
+
+        IntegrationRevision encryptedIntegrationRevision = encryptionSupport.encrypt(integrationRevision);
+
+        IntegrationRevision updatedIntegrationRevision = new IntegrationRevision.Builder()
+            .createFrom(encryptedIntegrationRevision)
+            .lastUpdated(rightNow)
+            .createdDate(rightNow)
+            .currentState(determineCurrentState(encryptedIntegrationRevision))
+            .stepsDone(new ArrayList<>())
+            .build();
+
+        return Creator.super.create(sec, updatedIntegrationRevision);
+    }
+
+
+    @Path("/{id}/{version}")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public void update(@NotNull @PathParam("id") @ApiParam(required = true) String id, @NotNull @PathParam("version") @ApiParam(required = true) String version, @ConvertGroup(from = Default.class, to = AllValidations.class) IntegrationRevision integrationRevision) {
```

Comment:
Does the UI even need to update?  I thought all updates were done by controllers.  Perhaps we should disallow REST updates.

Reply:
Right! A revision/deployment once created should be immutable. Only the state stuff should be allowed to change and that is something that is only handled internally by the controllers.